src/include/grpc_transcoding/path_matcher.h

```diff
@@ -223,8 +223,12 @@ inline int hex_digit_to_int(char c) {
 // If the next three characters are an escaped character then this function will
 // also return what character is escaped.
 bool GetEscapedChar(const std::string& src, size_t i,
-                    bool unescape_reserved_chars, bool unescape_slash_char,
-                    char* out) {
+                    UrlUnescapeSpec unescape_spec, char* out) {
+  const bool unescape_slash_char =
```

Comment:
the code readability is bad,  you are converting an enum type into two bool,  inside the loop, can we just use switch() on the spec.  and handle it accordingly.  the code will be much easier to read

Reply:
great point, addressed